src/Chapter3.hs

```diff
@@ -970,7 +1061,25 @@ implement the following functions:
 
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
-
+data Weekday = MON | TUE | WED | THU | FRI | SAT | SUN deriving (Enum, Show, Eq)
+instance Bounded Weekday where
+  minBound = MON
+  maxBound = SUN
+
+isWeekend :: Weekday -> Bool
+isWeekend a | (a == SAT) || (a == SUN) = True
+            | otherwise = False
+
+nextDay :: Weekday -> Weekday
+nextDay a | a == maxBound = minBound
+          | otherwise = succ a
+
+daysToParty :: Weekday -> Int
+daysToParty a = go a 0
+  where
+    go :: Weekday -> Int -> Int
+    go a b | a == FRI = b
+           | otherwise = go (nextDay a) (b + 1)   
```

Comment:
That's the correct implementation! Though, you can implement slightly more efficiently by using the `fromEnum` method of the `Enum` typeclass